packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  bool _atLineStart = true;
+  bool _hasMultipleLines = false;
+
+  /// Whether the string being built already has more than 1 line.
+  bool get hasMultipleLines => _hasMultipleLines;
+
+  /// Write text ensuring the specified prefixes for the first and subsequent
+  /// lines.
+  void write(String s) {
+    if (s.isEmpty)
+      return;
+
+    if (s == '\n') {
+      // Edge case to avoid adding trailing whitespace when the caller did
+      // not explicitly add trailing trailing whitespace.
+      if (_buffer.isEmpty) {
+        _buffer.write(prefixLineOne.trimRight());
+      } else if (_atLineStart) {
+        _buffer.write(prefixOtherLines.trimRight());
+        _hasMultipleLines = true;
+      }
+      _buffer.write('\n');
+      _atLineStart = true;
+      return;
+    }
+
+    if (_buffer.isEmpty) {
+      _buffer.write(prefixLineOne);
+    } else if (_atLineStart) {
+      _buffer.write(prefixOtherLines);
+      _hasMultipleLines = true;
+    }
+    bool lineTerminated = false;
+
+    if (s.endsWith('\n')) {
+      s = s.substring(0, s.length - 1);
+      lineTerminated = true;
+    }
+    final List<String> parts = s.split('\n');
+    _buffer.write(parts[0]);
+    for (int i = 1; i < parts.length; ++i) {
+      _buffer
+        ..write('\n')
+        ..write(prefixOtherLines)
+        ..write(parts[i]);
+    }
+
+    if (lineTerminated)
+      _buffer.write('\n');
+
+    _atLineStart = lineTerminated;
+  }
+
+  /// Write text assuming the text already obeys the specified prefixes for the
+  /// first and subsequent lines.
+  void writeRaw(String text) {
+    if (text.isEmpty)
+      return;
+    _buffer.write(text);
+    _atLineStart = text.endsWith('\n');
+  }
+
+
+  /// Write a line assuming the line obeys the specified prefixes. Ensures that
+  /// a newline is added if one is not present.
+  /// The same as [writeRaw] except a newline is added at the end of [line] if
+  /// one is not already present.
+  ///
+  /// A new line is not added if the input string already contains a newline.
+  void writeRawLine(String line) {
+    if (line.isEmpty)
+      return;
+    _buffer.write(line);
+    if (!line.endsWith('\n'))
+      _buffer.write('\n');
+    _atLineStart = true;
+  }
+
+  @override
+  String toString() => _buffer.toString();
+}
+
+/// Defines diagnostics data for an [Object].
+///
+/// DiagnosticsNode provides a high quality multi-line string dump via
+/// [toStringDeep]. The core members are the [name], [header], [getProperties],
+/// [object], and [getChildren]. All other members exist typically to provide
+/// hints for how [toStringDeep] and debugging tools should format output.
+abstract class DiagnosticsNode {
+  DiagnosticsNode({
+    @required this.name,
+    @required this.header,
+    this.style: DiagnosticsTreeStyle.normal,
+    bool hidden: false,
+    this.showNull: false,
+    this.showName: true,
+    this.showSeparator: true,
+    this.emptyDescription,
+    this.defaultValue,
+  })
```

Comment:
Yeah, like that. If the style guide says otherwise please submit a fix, it's a mistake. :-)

(As a general rule, if something is done consistently across the codebase, it trumps the style guide unless the style guide admits that it was changed post-hoc and that we intend to fix everything to match the new rule.)